src/cascadia/UIMarkdown/MarkdownToXaml.cpp

```diff
@@ -0,0 +1,451 @@
+{
+    cmark_node* parent;
+    cmark_node* grandparent;
+    bool tight;
+
+    bool entering = (ev_type == CMARK_EVENT_ENTER);
+
+    switch (cmark_node_get_type(node))
+    {
+    case CMARK_NODE_DOCUMENT:
+        break;
+
+    case CMARK_NODE_BLOCK_QUOTE:
+
+        // It's non-trivial to deal with the right-side vertical lines that
+        // we're accustomed to seeing for block quotes in markdown content.
+        // RichTextBlock doesn't have a good way of adding a border to a
+        // paragraph, it would seem.
+        //
+        // We could add a InlineUIContainer, with a Border in there, then
+        // put a new RichTextBlock in there, but I believe text selection
+        // wouldn't transit across the border.
+
+        // Instead, we're just going to add a new layer of indenting.
+
+        if (entering)
+        {
+            _EndParagraph();
+            _indent++;
+            _blockQuoteDepth++;
+        }
+        else
+        {
+            _EndParagraph();
+            _indent = std::max(0, _indent - 1);
+            _blockQuoteDepth = std::max(0, _blockQuoteDepth - 1);
+        }
+
+        break;
+
+    case CMARK_NODE_LIST:
+    {
+        // when `node->as.list.list_type == CMARK_BULLET_LIST`, we're an unordered list.
+        // Otherwise, we're an ordered one (and we might not start at 0).
+        //
+        // However, we don't support numbered lists for now.
+        if (entering)
+        {
+            _EndParagraph();
+            _indent++;
+        }
+        else
+        {
+            _EndParagraph();
+            _indent = std::max(0, _indent - 1);
+        }
+        break;
+    }
+
+    case CMARK_NODE_ITEM:
+        // A list item, either for a ordered list or an unordered one.
+        if (entering)
+        {
+            _EndParagraph();
+            _CurrentParagraph();
+            _NewRun().Text(winrt::hstring{ bullets[std::clamp(_indent - _blockQuoteDepth - 1, 0, 2)] });
+        }
+        break;
+
+    case CMARK_NODE_HEADING:
+        _EndParagraph();
+
+        // At the start of a header, change the font size to match the new
+        // level of header we're at. The text will come later, in a
+        // CMARK_NODE_TEXT
+        if (entering)
+        {
+            const auto level = cmark_node_get_heading_level(node);
+            _CurrentParagraph().FontSize(std::max(16u, 36u - ((level - 1) * 6u)));
+        }
+        break;
+
+    case CMARK_NODE_CODE_BLOCK:
+    {
+        _EndParagraph();
+        std::string_view code{ cmark_node_get_literal(node) };
+        const auto codeHstring{ winrt::hstring{ til::u8u16(code) } };
+
+        auto codeBlock = winrt::make<winrt::Microsoft::Terminal::UI::Markdown::implementation::CodeBlock>(codeHstring);
+        WUX::Documents::InlineUIContainer codeContainer{};
+        codeContainer.Child(codeBlock);
+        _CurrentParagraph().Inlines().Append(codeContainer);
+
+        _EndParagraph();
+        _CurrentParagraph();

```

Comment:
is this necessary? same Q about using it for its side effect

Reply:
_why is this load bearing_? I can't seem to remove this call without causing the parser to always crash. Weird.